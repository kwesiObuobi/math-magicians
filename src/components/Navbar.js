import React from 'react';

const Navbar = () => (
  <div>
    <p>This is the navbar</p>
    <p>It has 2 p tags</p>
  </div>
);

export default Navbar;
