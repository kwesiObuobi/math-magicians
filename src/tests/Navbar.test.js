import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('Navbar Component', () => {
  it('renders the Navbar component', () => {
    render(<BrowserRouter><Navbar /></BrowserRouter>);
    expect(screen.getByText('Math Magicians')).toBeInTheDocument();
    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.queryByRole('span')).toBeNull();
  });

  it('should correctly render the Navbar component', () => {
    const tree = renderer.create(<BrowserRouter><Navbar /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
