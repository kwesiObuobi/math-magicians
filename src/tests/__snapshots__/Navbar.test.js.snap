// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should correctly render the Navbar component 1`] = `
<div
  className="nav-container"
>
  <nav
    className="navbar"
  >
    <h1
      className="logo"
    >
      Math Magicians
    </h1>
    <ul
      className="nav-links"
    >
      <li>
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="nav-link"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className="nav-link"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
